gms/gossiper.cc

```diff
@@ -1138,7 +1134,7 @@ int64_t gossiper::get_endpoint_downtime(inet_address ep) const noexcept {
 // It is called from failure_detector
 future<> gossiper::convict(inet_address endpoint) {
     auto* state = get_endpoint_state_for_endpoint_ptr(endpoint);
-    if (!state || !state->is_alive()) {
```

Comment:
Why get rid of the check? (shouldn't we use `is_alive(endpoint)` now instead?)

Reply:
Later version of this series brought the check back.
It is indeed needed to prevent convict from repeatedly calling `mark_dead` for the endpoint.